core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -977,10 +977,9 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
 
                   case Some(epochAndMetadata) =>
                     if (epochAndMetadata.coordinatorEpoch == coordinatorEpoch) {
-                      // This was attempted epoch fence that failed, so mark this state on the metadata
-                      epochAndMetadata.transactionMetadata.hasFailedEpochFence = true
+                      // For TV2, we allow re-bumping the epoch on retry, so don't set hasFailedEpochFence = true
```

Comment:
For this comment, should we say that we don't complete the epoch bump so we can simply retry the epoch bump.